c10/core/TensorImpl.cpp

```diff
@@ -93,20 +96,28 @@ void TensorImpl::release_resources() {
 }
 
 int64_t TensorImpl::dim() const {
+  TORCH_CHECK(type_id_ != UndefinedTensorId(), type_id_, " does not have dim");
   return sizes_.size();
 }
 
 int64_t TensorImpl::size(int64_t d) const {
+  TORCH_CHECK(type_id_ != UndefinedTensorId(), type_id_, " does not have size");
   d = at::maybe_wrap_dim(d, dim(), false);
   return sizes_[d];
 }
 
 int64_t TensorImpl::stride(int64_t d) const {
+  TORCH_CHECK(type_id_ != UndefinedTensorId() && layout() == kStrided, type_id_, " does not have stride");
   d = at::maybe_wrap_dim(d, dim(), false);
   return strides_[d];
 }
 
 TensorImpl* TensorImpl::maybe_zero_dim(bool condition_when_zero_dim) {
+  // We only allow this operation on dense tensors
+  if (!(has_storage() && layout() == kStrided)) {
+    TORCH_CHECK(condition_when_zero_dim == (dim() == 0),
```

Comment:
I know you just moved this, but I think this should be an INTERAL_CHECK; this isn't a public API.

Reply:
Fixed.